be/src/common/configbase.h

```diff
@@ -93,60 +104,154 @@ inline std::ostream& operator<<(std::ostream& os, const MutableString& s) {
+
+    const char* defval() const { return _defval; }
 
+    bool valmutable() const { return _valmutable; }
+
+    bool set_value(std::string value);
+
+    virtual std::string value() const = 0;
+
+    static void clear_fields() { _s_field_map.clear(); }
+
+    static std::map<std::string, Field*>& fields() { return _s_field_map; }
+
+    static std::optional<Field*> get(const std::string& name_or_alias);
+
+protected:
+    inline static std::map<std::string, Field*> _s_field_map{};
+
+    virtual bool parse_value(const std::string& value) = 0;
+
+    const char* _type;
+    const char* _name;
+    void* _storage;
+    const char* _defval;
+    bool _valmutable;
+};
+
+template <typename T, typename = void>
+class FieldImpl;
+
+template <typename T>
+class FieldImpl<T> final : public Field {
 public:
-    static std::map<std::string, Field>* _s_field_map;
+    FieldImpl(const char* type, const char* name, void* storage, const char* defval, bool valmutable)
+            : Field(type, name, storage, defval, valmutable) {}
+
+    std::string value() const override { return fmt::format("{}", *reinterpret_cast<T*>(_storage)); }
 
+    bool parse_value(const std::string& valstr) override { return strtox(valstr, *reinterpret_cast<T*>(_storage)); }
+};
+
+//// FieldImpl<std::vector<T>>
+template <typename T>
+class FieldImpl<std::vector<T>> final : public Field {
+public:
+    FieldImpl(const char* type, const char* name, void* storage, const char* defval, bool valmutable)
+            : Field(type, name, storage, defval, valmutable) {}
+
+    std::string value() const override {
+        auto as_str = [](const T& v) { return fmt::format("{}", v); };
+        const auto& v = *reinterpret_cast<const std::vector<T>*>(_storage);
+        return JoinMapped(v, as_str, ",");
+    }
+
+    bool parse_value(const std::string& valstr) override {
+        std::vector<T> tmp;
+        std::vector<std::string> parts = strings::Split(valstr, ",");
+        for (auto& part : parts) {
+            T v;
+            StripWhiteSpace(&part);
+            if (part.empty()) {
+                continue;
+            }
+            if (!strtox(part, v)) {
+                return false;
+            }
+            tmp.emplace_back(std::move(v));
+        }
+        auto& value = *reinterpret_cast<std::vector<T>*>(_storage);
+        value.swap(tmp);
+        return true;
+    }
+};
+
+class Alias {
 public:
-    Register(const char* ftype, const char* fname, void* fstorage, const char* fdefval, bool fvalmutable) {
-        if (_s_field_map == nullptr) {
-            _s_field_map = new std::map<std::string, Field>();
+    explicit Alias(const char* alias, Field* field) {
+        assert(strcmp(field->name(), alias) != 0);
+        [[maybe_unused]] auto [_, ok] = Field::fields().emplace(std::string(alias), field);
+        if (!ok) {
+            std::cerr << fmt::format("The alias '{}' for config '{}' already exists\n", alias, field->name());
```

Comment:
it should be the `alias` name already used, but may not alias to the config. e.g.

```
CONF_Alias(config_a, alias_xxx);
CONF_Alias(config_b, alias_xxx);
```

Reply:
for a second thought, it looks like impossible to have the same alias defined to different config item. the Alias object variable will be the same, which will be a compiling error.

Reply:
> for a second thought, it looks like impossible to have the same alias defined to different config item. the Alias object variable will be the same, which will be a compiling error.

Good catch!

Reply:
Fixed, PTAL